pkg/controller/clusterprovision/clusterprovision_controller.go

```diff
@@ -173,7 +173,13 @@ func (r *ReconcileClusterProvision) Reconcile(request reconcile.Request) (reconc
 			return r.reconcileRunningJob(instance, pLog)
 		}
 		return r.transitionStage(instance, hivev1.ClusterProvisionStageFailed, "NoJobReference", "Missing reference to install job", pLog)
-	case hivev1.ClusterProvisionStageComplete, hivev1.ClusterProvisionStageFailed:
+	case hivev1.ClusterProvisionStageComplete:
+		pLog.Debugf("ClusterProvision is %s", instance.Spec.Stage)
+		if instance.Status.JobRef != nil && time.Since(instance.CreationTimestamp.Time) > (24*time.Hour) {
+			return r.deleteInstallJob(instance, pLog)
+		}
+		return reconcile.Result{}, nil
```

Comment:
If the install job needs to be deleted in the future, the `reconcile.Result` should have `RequeueAfter` set to the duration remaining until the delete.